src/cpp/r/RCntxt.cpp

```diff
@@ -111,12 +111,24 @@ SEXP RCntxt::originalFunctionCall() const
 
 Error RCntxt::fileName(std::string* pFileName) const
 {
-   if (srcref() && TYPEOF(srcref()) != NILSXP)
+   // skip over bytecode srcrefs
+   RCntxt context = *this;
+   SEXP ref = context.srcref();
+   while (ref && isByteCodeSrcRef(ref))
```

Comment:
I think we're safe here because all entries into the bytecode interpreter typically run through an `Rf_applyClosure()` call; e.g. we'll typically see callstacks like:

```
#24 0x00007f36959e2c44 in bcEval () from /.../lib/libR.so
#25 0x00007f36959d8275 in Rf_eval () from /.../lib/libR.so
#26 0x00007f36959d957c in Rf_applyClosure () from /.../lib/libR.so
```

And that `Rf_applyClosure()` call will generate a context tied to the same bytecode evaluation that we can use for the debugger. That said, it looks like it is possible for `bcEval()` to be called recursively, but it still seems like that will only happen in a context where a separate non-bytecode context is on the stack.